### PR TITLE
Persistent state using sled

### LINKED ISSUE #24: Scheduler state persistance
We may want to consider some offline storage based on an in-memory database. This would allow for better handling of state in case of trouble.

By this I mean something like tracking allocations, timing data, etc.

Let's use this issue to allocate time to discuss and define a way forward.

### PR DESCRIPTION
**Pull request by neithanmo**
this adds:
- a sled dependency which is used to implement a DB interface
- adds the logic to store and retrieve the scheduler state from a previous session using the info stored in the DB
- other improvements

closes #24

### CODE REVIEW

File: scheduler/src/db.rs! [hunk 1]
```diff
@@ -0,0 +1,98 @@
+use crate::{Error, Result};
```
Comment (jleni) [inline scheduler/src/db.rs!:1]: It is file correct?  (notice ! at the end)

File: scheduler/src/scheduler.rs [hunk 2]
```diff
@@ -11,17 +11,20 @@ use std::time::Instant;
 use tracing::{debug, error, instrument, warn};
 
 use crate::config::{Settings, Task};
+use crate::db::DataBase;
```
Comment (jleni) [inline scheduler/src/scheduler.rs:14]: I think it would be better to call this `Database`

File: scheduler/src/server.rs [hunk 3]
```diff
@@ -1,4 +1,4 @@
-use std::result::Result;
+//use std::result::Result;
```
Comment (jleni) [inline scheduler/src/server.rs:1]: Should this be removed?

**Review by jleni — CHANGES_REQUESTED** [review]
very minor nitpicking

File: scheduler/src/db.rs! (hunk 1 above)
Comment (neithanmo) [reply scheduler/src/db.rs!:1]: yes, this file should be removed.

**Pull request merged**